src/Adapter/OrderReturnState/Repository/OrderReturnStateRepository.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\OrderReturnState\Repository;
+
+use OrderReturnState;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturnState\Exception\OrderReturnStateNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturnState\ValueObject\OrderReturnStateId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\OrderReturnState\Repository\OrderReturnStateRepositoryInterface;
+use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
+use PrestaShopException;
+
+class OrderReturnStateRepository extends AbstractObjectModelRepository implements OrderReturnStateRepositoryInterface
+{
+    /**
+     * Gets legacy OrderReturnState
+     *
+     * @param OrderReturnStateId $orderReturnStateId
+     *
+     * @return OrderReturnState
+     *
+     * @throws CoreException
+     * @throws OrderReturnStateNotFoundException
+     */
+    public function get(OrderReturnStateId $orderReturnStateId): OrderReturnState
```

Comment:
Should return an object an `OrderReturnStateInterface` that implements all the getters , so we don't have legacy code in the new code.

Reply:
Same comment as previous repository, these repositories are not meant to return a core interface but ObjectModel instances